vault/resource_jwt_auth_backend_role.go

```diff
@@ -367,6 +384,14 @@ func jwtAuthBackendRoleDataToWrite(d *schema.ResourceData, create bool) map[stri
 	data["bound_audiences"] = util.TerraformSetToStringArray(d.Get("bound_audiences"))
 	data["user_claim"] = d.Get("user_claim").(string)
 
+	if v, ok := d.GetOkExists("user_claim_json_pointer"); ok {
```

Comment:
I couldn't find a way around using `GetOkExists`. Looks like it's deprecated but there isn't a good solution to the problem yet from my search. Let me know if there is something you all are typically doing for these optional boolean params.

Reply:
We have used `GetOkExists` in the past due to `GetOk` not behaving exactly the way we'd like when it comes to boolean parameters. I think for this case it is fine to go with `GetOkExists` similar to other fields within this file. `d.Get` would also work I believe

Reply:
If your field type is a boolean and has a default then using `GetOk()` should work the same as `GetOkExists()`

Reply:
The `user_claim_json_pointer` currently has a default. With that, my experience is that `GetOk()` and `GetOkExists()` don't actually behave the same. For example, if `user_claim_json_pointer=true` on create, then we switch it to `false` for an update, the `false` update is not applied when using `GetOk()` but is applied when using `GetOkExists()`. We want the `GetOkExists()` behavior in this case.

I tried a number of different things and wasn't able to find a workaround that allows me to not use `GetOkExists()`.

Reply:
Perhaps we could just use `Get()`.

Reply:
Yeah, the behavior of `GetOk()` and `GetOkExists()` are a bit odd IMO. Since you have a default then using `Get()` should work fine.

Reply:
I changed this to use `Get()` in [749ca8b](https://github.com/hashicorp/terraform-provider-vault/pull/1478/commits/749ca8bd694f3d27a497158201166c6a1d934abf). That works as expected with the default.